dev/tests/integration/testsuite/Magento/Framework/View/Layout/Reader/BlockTest.php

```diff
@@ -0,0 +1,109 @@
+<?php
+
+namespace Magento\Framework\View\Layout\Reader;
+
+class BlockTest extends \PHPUnit_Framework_TestCase
+{
+    const IDX_TYPE = 0;
+    const IDX_PARENT = 2;
+
+    /**
+     * @var Block
+     */
+    private $block;
+
+    /**
+     * @var Context
+     */
+    private $readerContext;
+
+    private $blockName = 'test.block';
+    private $childBlockName = 'test.child.block';
+
+    public function setUp()
+    {
+        $this->block = \Magento\TestFramework\Helper\Bootstrap::getObjectManager()->create(
+            \Magento\Framework\View\Layout\Reader\Block::class
+        );
+        
+        $this->readerContext = \Magento\TestFramework\Helper\Bootstrap::getObjectManager()->create(
+            \Magento\Framework\View\Layout\Reader\Context::class
+        );
+    }
+
+    public function testInterpretBlockDirective()
+    {
+        $pageXml = new \Magento\Framework\View\Layout\Element(__DIR__ . '/_files/_layout_update_block.xml', 0, true);
+        $parentElement = new \Magento\Framework\View\Layout\Element('<page></page>');
+
+        foreach ($pageXml->xpath('body/block') as $blockElement) {
+            $this->assertTrue(in_array($blockElement->getName(), $this->block->getSupportedNodes()));
+
+            $this->block->interpret($this->readerContext, $blockElement, $parentElement);
+        }
+
+        $structure = $this->readerContext->getScheduledStructure();
```

Comment:
Better to not testing ScheduledStructure in this test. This is private representation of layout and should not used outside